src/app/common/use-wallet-type.ts

```
@@ -1,24 +1,18 @@
 import { useMemo } from 'react';
 
-import { useHasLedgerKeys } from '@app/store/ledger/ledger.selectors';
-import { useCurrentKeyDetails } from '@app/store/software-keys/software-key.selectors';
+import { type WalletType, useWalletTypeSelector } from '@app/store/common/wallet-type.selectors';
 
-enum WalletType {
-  Ledger = 'ledger',
-  Software = 'software',
-}
-
-function isLedgerWallet(walletType: WalletType) {
+function isLedgerWallet(walletType?: WalletType) {
   return walletType === 'ledger';
 }
 
-function isSoftwareWallet(walletType: WalletType) {
+function isSoftwareWallet(walletType?: WalletType) {
   return walletType === 'software';
 }
 
 type WalletTypeMap<T> = Record<WalletType, T>;
 
-function whenWallet(walletType: WalletType) {
+function whenWallet(walletType?: WalletType) {
```

Maybe we can investigate adding another `WalletType` for when the user has not setup the wallet yet. 

We throw an error now on `Sign Out` in `whenWallet` just below here 

`Wallet is neither of type `ledger` nor `software``. We have a bug for it but I couldn't find it just now.

---

this could also just fail silently? 🤔

---

I think having it set to undefined is fined, basically means there is not wallet setup now. It's more of an issue with how we handle it. I'd open another issue for that so we can address that separately

---

Here it is: https://github.com/leather-wallet/issues/issues/74